Checkr/src/components/molecules/table/index.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState } from 'react';
+import { Table, TableContainer, TableHead, TableBody, TableRow, TableCell, Paper } from '@mui/material';
+import { Dialog, DialogTitle, DialogContent } from '@mui/material';
+import Chips from '../../../components/atoms/Chips/index';
+import theme from '../../../components/themes/index';
+import { ThemeProvider } from '@mui/system';
+import SearchIcon from '@mui/icons-material/Search';
+import FilterListIcon from '@mui/icons-material/FilterList';
+import { MoreVertOutlined } from '@mui/icons-material';
+import IconButton from '@mui/material/IconButton';
+import Stack from '@mui/material/Stack';
+import Textfield from '../../../components/atoms/TextField/index';
+import InputAdornment from '@mui/material/InputAdornment';
+import Buttons from '../../../components/atoms/Button/index';
+import { Box } from '@mui/system';
+import CheckboxBox from '../../../components/organisms/FilterDialog/index';
+import AtomTypography from '../../atoms/Typography';
+import { Link } from 'react-router-dom';
+
+
+export interface TableData {
+  id: number;
+  name: string;
+  adjusdiction: string;
+  status: {
+    label: string;
+    //color: string;
+  };
+  location: string;
+  date: string;
+}
+
+interface TableMoleculeProps {
+  headerLabels: string[];
+  tableData: TableData[];
+  // onRowClick: (rowData: TableData) => void;
+
+}
+
+const TableMolecule: React.FC<TableMoleculeProps> = (props) => {
+  const { headerLabels, tableData } = props;
+  const [searchValue, setSearchValue] = useState('');
+  const [openFilter, setopenFilter] = useState(false);
+  const [filteredData, setFilteredData] = useState(tableData);
+
+  const [checkedItems, setcheckedItems] = useState({
+    allStatus: true,
+    clear: false,
+    consider: false,
+  });
+
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.toLowerCase();
+    setSearchValue(value);
+
+    const filteredTableData = tableData.filter((row) => row.name.toLowerCase().includes(value)
+    );
+    setFilteredData(filteredTableData);
+  };
+
+  const handleFilterClick = () => {
+    setopenFilter(true);
```

Comment:
No need for an explicit function to call the setState function, directly call the setState function.